src/charm.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm for Vault running on Kubernetes.
+
+For more information on Vault, please visit https://www.vaultproject.io/.
+"""
+
+import json
+import logging
+import os
+
+from charms.tls_certificates_interface.v0.tls_certificates import (
+    Cert,
+    InsecureCertificatesProvides,
+)
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus
+from ops.pebble import Layer
+
+import interface_vault_operator_peers
+from vault import Vault
+
+logger = logging.getLogger(__name__)
+
+
+STORAGE_PATH = "/var/lib/juju/storage/vault-storage/0"
```

Comment:
Is this not `/srv` as per line 20 in `metadata.yaml`?

Reply:
Indeed this is a bit strange. This variable is used to change ownership of a certain path to "Fix up storage permissions (broken on CDK on AWS otherwise)'." but it isn't the backend path used by Vault. @ChrisMacNaughton Can you provide a bit more context here?

Reply:
I created a new variable for the /srv/ path and a new method to change the ownership of the path so that that the confusion isn't hidden too far in the code...I'll remove this once we properly understand what's this for.

Reply:
Fixed now. We're using /srv